lib/Utilities/NameValidator.cpp

```diff
@@ -170,12 +156,34 @@ bool CollectionNameValidator::isAllowedName(bool allowSystem,
     }
   }
 
-  if (extendedNames &&
-      !velocypack::Utf8Helper::isValidUtf8(
-          reinterpret_cast<std::uint8_t const*>(name.data()), name.size())) {
+  if (extendedNames && !name.empty()) {
+    unsigned char c = static_cast<unsigned char>(name[0]);
```

Comment:
Why unsigned char instead of just char?
Btw name.front

Reply:
Fixed